pkg/controllermanager/controller/project/project_control.go

```diff
@@ -44,6 +45,15 @@ func (c *Controller) projectAdd(obj interface{}) {
 }
 
 func (c *Controller) projectUpdate(oldObj, newObj interface{}) {
+	var (
+		oldProject = oldObj.(*gardenv1beta1.Project)
+		newProject = newObj.(*gardenv1beta1.Project)
+	)
+
+	if apiequality.Semantic.DeepEqual(oldProject.Generation, newProject.Generation) {
```

Comment:
```suggestion
	if newProject.Generation == newProject.Status.ObservedGeneration {
```

Reply:
Actually we can check for equality on the newProject vs oldProject objects on update. That way we can get rid of the observedGeneration in the status, if it doesn't introduce any issues (API incompatibility) on existing resources that are being fetch from etcd. @timuthy what are your thoughts about this?

Reply:
Generally I think that comparing against `ObservedGeneration` is better. But lets wait for @timuthy.

`ObservedGeneration` is incremented by the gardener controller manager only if reconciliation is successful. 

`Generation` gets incremented by the gardener apiserver when an update request is processed for the corresponding resource and only if there is a difference between the specs of the previous and new resources.

Reply:
Technically, we can also call `Semantic.DeepEqual(oldProject, newProject)` but I think we don't benefit from that.

The `Generation` is already increased if  `Semantic.DeepEqual(oldProject, newProject) == false` (see https://github.com/gardener/gardener/blob/3501ea171f3963edcf88f7afa7c73aa6c7ea89ac/pkg/registry/garden/project/strategy.go#L61) and already belongs to the API which is why we also shouldn't remove it now (compatibility). I think it's also considered a Kubernetes best practice to have an `ObservedGeneration` when the resource has a `status` sub-resource.

Bottom line, `newProject.Generation == newProject.Status.ObservedGeneration` is probably cheaper and simpler in this case.